### PR TITLE
[Optimize](row store) optimize serialization and deserialization

### PR DESCRIPTION
**Pull request by eldenmoon**
1. get DataTypeSerde in advance to avoid get temporary DataTypeSerde  iterate each column
2. iterate the original row once is enoungh for deserializing, introdcue a map for record the index of each column's unique id

# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by eldenmoon** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 33.49 seconds
 stream load tsv:          429 seconds loaded 74807831229 Bytes, about 166 MB/s
 stream load json:         24 seconds loaded 2358488459 Bytes, about 93 MB/s
 stream load orc:          60 seconds loaded 1101869774 Bytes, about 17 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230516081754_clickbench_pr_144462.html

**Comment by eldenmoon** [comment]
run buildall

File: be/src/exec/rowid_fetcher.cpp [hunk 1]
```diff
@@ -136,10 +138,16 @@ Status RowIDFetcher::_merge_rpc_results(const PMultiGetRequest& request,
             if (output_block->is_empty_column()) {
                 *output_block = vectorized::Block(_fetch_option.desc->slots(), 1);
             }
+            if (serdes.empty() && col_uid_to_idx.empty()) {
```
Comment (amorynan) [inline be/src/exec/rowid_fetcher.cpp:141]: is possible here serde is not empty but col_uid_to_idx empty ?

Comment (eldenmoon) [reply be/src/exec/rowid_fetcher.cpp:141]: that is impossible

**Comment by amorynan** [comment]
LGTM

**Review by qidaye — APPROVED** [review]
LGTM

**Pull request merged**